packages/ra-core/src/controller/CreateController.tsx

```diff
@@ -36,11 +36,22 @@ interface Props {
     resource: string;
 }
 
-interface EnhancedProps {
-    crudCreate: Dispatch<typeof crudCreateAction>;
-    isLoading: boolean;
-    translate: Translate;
-}
+export const getRecord = ({ state, search }, record: any = {}) =>
```

Comment:
this should live in the reducers directory as it's a selector (same with the test of this function)

Reply:
Agreed, I will extract selectors